pkg/apis/keycloak/v1alpha1/keycloakclient_types.go

```diff
@@ -13,6 +13,11 @@ type KeycloakClientSpec struct {
 	// Keycloak Client REST object.
 	// +kubebuilder:validation:Required
 	Client *KeycloakAPIClient `json:"client"`
+	// Client Roles
+	// +optional
+	// +listType=map
```

Comment:
Could we use an atomic list here?

Reply:
My reasoning was that the reconciler code I wrote treats it like map: It identifies roles by their names when figuring out what to create/update/delete and just assumes there aren't any duplicates (which the map type would ensure). The mergeability is just a nice side effect.

Reply:
I believe we need a rebase here.